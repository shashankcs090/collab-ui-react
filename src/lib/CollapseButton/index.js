import React from 'react';
import PropTypes from 'prop-types';
import { Button, Icon } from '@collab-ui/react';

const CollapseButton = props => {
  const { collapse, alignment, onClick, className, ...otherProps } = props;

  const handleClick = () => {
    onClick && onClick();
  };

  const getIconName = () => {
    return collapse ? 'arrow-right-optical_14' : 'arrow-left-optical_14';
  };

  return (
    <Button
      ariaLabel={collapse ? 'expand' : 'collapse'}
      className={
        'cui-collapse-button' +
        ` cui-collapse-button--${alignment}` +
        `${(className && ` ${className}`) || ''}`
      }
      children={<Icon name={getIconName()}/>}
      onClick={() => handleClick()}
      {...otherProps}
    />
  );
};

CollapseButton.displayName = 'CollapseButton';

CollapseButton.defaultProps = {
  collapse: true,
  alignment: 'left',
  onClick: null,
  className: '',
};

CollapseButton.propTypes = {
  collapse: PropTypes.bool,
  alignment: PropTypes.oneOf(['left', 'right', 'top', 'bottom']),
  onClick: PropTypes.func,
  className: PropTypes.string,
};

export default CollapseButton;
