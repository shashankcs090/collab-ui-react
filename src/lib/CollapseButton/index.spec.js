import React from 'react';
import { mount, shallow } from 'enzyme';
import CollapseButton from '../CollapseButton';

describe('tests for <CollapseButton />', () => {
  it('should match SnapShot', () => {
    const container = shallow(<CollapseButton/>);

    expect(container).toMatchSnapshot();
  });

  it('should call onClick callback when the button is clicked', () => {
    const onClick = jest.fn();
    const container = mount(<CollapseButton onClick={onClick} />);
    container.find('.cui-button').simulate('click');

    expect(onClick).toHaveBeenCalled();
  });

  it('when collapse is true it should display right icon and label', () => {
    const container = shallow(<CollapseButton collapse={true} />);
    expect(container.find('Icon').props().name).toEqual('arrow-right-optical_14');
    expect(container.find('Button').props().ariaLabel).toEqual('expand');
  });

  it('when collapse is false it should display right icon and label', () => {
    const container = shallow(<CollapseButton collapse={false} />);
    expect(container.find('Icon').props().name).toEqual('arrow-left-optical_14');
    expect(container.find('Button').props().ariaLabel).toEqual('collapse');
  });

  describe('should apply correct alignment classes', () => {

    it('when alignment is left', () => {
      const container = shallow(<CollapseButton alignment={'left'} />);
      expect(container.find('.cui-collapse-button').hasClass('cui-collapse-button--left')).toEqual(true);
    });

    it('when alignment is right', () => {
      const container = shallow(<CollapseButton alignment={'right'} />);
      expect(container.find('.cui-collapse-button').hasClass('cui-collapse-button--right')).toEqual(true);
    });

    it('when alignment is top', () => {
      const container = shallow(<CollapseButton alignment={'top'} />);
      expect(container.find('.cui-collapse-button').hasClass('cui-collapse-button--top')).toEqual(true);
    });

    it('when alignment is down', () => {
      const container = shallow(<CollapseButton alignment={'bottom'} />);
      expect(container.find('.cui-collapse-button').hasClass('cui-collapse-button--bottom')).toEqual(true);
    });
  });

});
