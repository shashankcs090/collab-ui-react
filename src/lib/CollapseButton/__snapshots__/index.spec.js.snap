// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <CollapseButton /> should match SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CollapseButton
    alignment="left"
    className=""
    collapse={true}
    onClick={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "active": false,
      "ariaLabel": "expand",
      "ariaLabelledBy": "",
      "children": <Icon
        className=""
        color=""
        isAria={true}
        name="arrow-right-optical_14"
      />,
      "circle": false,
      "className": "cui-collapse-button cui-collapse-button--left",
      "color": "",
      "containerLarge": false,
      "disabled": false,
      "expand": false,
      "href": "",
      "label": "",
      "large": false,
      "loading": false,
      "onClick": [Function],
      "style": Object {},
      "tag": "button",
      "type": "button",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "className": "",
        "color": "",
        "isAria": true,
        "name": "arrow-right-optical_14",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "active": false,
        "ariaLabel": "expand",
        "ariaLabelledBy": "",
        "children": <Icon
          className=""
          color=""
          isAria={true}
          name="arrow-right-optical_14"
        />,
        "circle": false,
        "className": "cui-collapse-button cui-collapse-button--left",
        "color": "",
        "containerLarge": false,
        "disabled": false,
        "expand": false,
        "href": "",
        "label": "",
        "large": false,
        "loading": false,
        "onClick": [Function],
        "style": Object {},
        "tag": "button",
        "type": "button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "",
          "color": "",
          "isAria": true,
          "name": "arrow-right-optical_14",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
